src/pages/CreateAd.js

```diff
@@ -0,0 +1,167 @@
+    inputPrice: "",
+    inputPaymentMethod: "",
+    inputCategory: "",
+    inputPhotos: [],
+    inputInstallments: "",
+    products: [],
+    inputSearchProduct: "",
+  };
+  
+  componentDidMount = () => {
+      this.getProducts()
+  }
+
+  getProducts = () => {
+    axios
+    .get(
+      "https://us-central1-labenu-apis.cloudfunctions.net/eloFourTwo/products",
+      body,
+      {
+        headers: {
+          Authorization: "joao-vitor-alves",
+        },
+      }
+    ).then((res) => {
+        this.setState({products: res.data})
+    })
+  }
+
+  createProduct = () => {
+    const body = {
+      name: this.state.inputName,
+      description: this.state.inputDescription,
+      price: this.state.inputPrice,
+      paymentMethod: this.state.inputPaymentMethod,
+      category: this.state.inputCategory,
+      photos: this.state.inputPhotos,
+      installments: this.state.inputInstallments,
+    };
+    axios
+      .post(
+        "https://us-central1-labenu-apis.cloudfunctions.net/eloFourTwo/products",
+        body,
+        {
+          headers: {
+            Authorization: "joao-vitor-alves",
+          },
+        }
+      )
+      .then((res) => {
+        this.setState({ inputName: "" });
+        this.setState({ inputDescription: "" });
+        this.setState({ inputPrice: "" });
+        this.setState({ inputPaymentMethod: "" });
+        this.setState({ inputCategory: "" });
+        this.setState({ inputPhotos: "" });
+        this.setState({ inputInstallments: "" });
+        alert("Seu produto já foi anunciado!")
+      })
+      .catch((err) => {
+        this.setState({ inputName: "" });
+        this.setState({ inputDescription: "" });
+        this.setState({ inputPrice: "" });
+        this.setState({ inputPaymentMethod: "" });
+        this.setState({ inputCategory: "" });
+        this.setState({ inputPhotos: "" });
+        this.setState({ inputInstallments: "" });
+        alert("Ocorreu um erro! Estamos trabalhando para solucioná-lo!")
+        console.log(err)
+      })
+  };
+
+  handleName = (e) => {
+    this.setState({ inputName: e.target.value });
+  };
+  handleDescription = (e) => {
+    this.setState({ inputDescription: e.target.value });
+  };
+  handlePrice = (e) => {
+    this.setState({ inputPrice: e.target.value });
+  };
+  handlePaymentMethod = (e) => {
+    this.setState({ inputPaymentMethod: e.target.value });
+  };
+  handleCategory = (e) => {
+    this.setState({ inputCategory: e.target.value });
+  };
+  handlePhotos = (e) => {
+    this.setState({ inputPhotos: e.target.value });
+  };
+  handleInstallments = (e) => {
+    this.setState({ inputInstallments: e.target.value });
+  };
+
+  deleteProduct = (productId) => {
+    axios.delete(`https://us-central1-labenu-apis.cloudfunctions.net/eloFourTwo/products/${productId}`, 
+    {
+        headers: {
+            Authorization: "joao-vitor-alves"
+        }
+    }).then((res) => {
+        alert("Esse produto foi deletado!")
```

Comment:
Depois que o produto foi deletado, tem que atualizar o this.state.products. Pode fazer isso usando direto o getProducts() que vai buscar os produtos da API, porém se quiser uma melhor fluídez para o usuário, pode deletar internamente direto no this.state.products.